Sources/Valkey/Connection/ValkeyChannelHandler+stateMachine.swift

```diff
@@ -65,11 +65,12 @@ extension ValkeyChannelHandler {
         @usableFromInline
         struct ConnectedState {
             let context: Context
-            var pendingHelloCommand: PendingCommand
+            let pendingHelloCommand: PendingCommand
+            var pendingCommands: Deque<PendingCommand>
 
-            func cancel(requestID: Int) -> PendingCommand? {
+            func cancel(requestID: Int) -> [PendingCommand]? {
                 if pendingHelloCommand.requestID == requestID {
-                    return pendingHelloCommand
+                    return [pendingHelloCommand] + pendingCommands
                 }
```

Comment:
I think the cancel code here should be unreachable. HELLO and SETINFO are triggered from within the channel handler.